warn-stealers.lua

```diff
@@ -0,0 +1,107 @@
+-- Script to warn when creatures that may steal food become visible
+--[====[
+warn-stealers
+=============
+Will make a zoomable announcement whenever a creature that can eat food, guzzle drinks, or steal items enters the map and moves into a revealed location.
+Takes ``start`` or ``stop`` as parameters.
+]====]
+
+local persistTable = require("persist-table")
+local eventful = require("plugins.eventful")
+local repeatUtil = require("repeat-util")
+
+local eventfulKey = "warn-stealers"
+local numTicksBetweenChecks = 100
+
+if df.global.gamemode ~= df.game_mode.DWARF then
+    if df.global.gamemode ~= df.game_mode.NONE then
+        -- errors when gamemode is NONE
+        persistTable.GlobalTable.warnStealersCache = nil
+    end
+    return
+end
+
+if not persistTable.GlobalTable.warnStealersCache then
+    persistTable.GlobalTable.warnStealersCache = {}
+end
+local cache = persistTable.GlobalTable.warnStealersCache
+
+local function isUnitHidden(unit)
+    local block = dfhack.maps.getTileBlock(unit.pos)
+    if not block then
+        return false
```

Comment:
i'm not sure when a unit would be in an invalid block, but shouldn't this return true to indicate that the unit is hidden?

Reply:
There is also `dfhack.units.isHidden()` which handles considerably more cases

Reply:
I tested units with negative coords and they were visible. And my testing showed me that it was the tile they were on that defined whether they were visible, but OK, since there're more cases, I'll use that.